manifests/params.pp

```diff
@@ -31,6 +31,12 @@
     $service_enable = true
   }
 
+  if ($::operatingsystem == 'Ubuntu' and $::operatingsystemrelease in ['15.04','15.10','16.04','16.10']) or ($::operatingsystem == 'Debian' and $::operatingsystemrelease == '8') {
```

Comment:
It's not a core fact unfortunately no. Adding a dependency is one option, but seems a bit heavy handed for just one fact. 

However, it could be added as a fact to this module with it's own namespace (eg. gitlab_systemd_check or something), using only the code for the systemd check from the linked fact:

```ruby
Facter.add(:gilab_systemd_check) do
  confine :kernel => :linux
  setcode do
    Facter::Core::Execution.exec('ps -p 1 -o comm=') == 'systemd'
  end
end
```
The benefit of adding the fact would be that the code becomes a lot more manageable:

```puppet
if ($::gilab_systemd_check) {
```